pkg/vmconfig/diskpromo/diskpromo_reconciler.go

```diff
@@ -187,8 +172,11 @@ func (r reconciler) Reconcile(
 	// not participating in snapshots.
 	for i := range allDisks {
 		d := allDisks[i].(*vimtypes.VirtualDisk)
-		if _, ok := snapshotDiskKeys[d.Key]; !ok {
-			if d.VDiskId == nil { // Skip FCDs
+
```

Comment:
If you want, you can reduce the indent a little.
```
for i := range allDisks {
		d := allDisks[i].(*vimtypes.VirtualDisk)
		if _, ok := snapshotDiskKeys[d.Key]; !ok {
			continue
		}
		if d.VDiskId != nil {
			continue
		}
```
(or even merge these two conditions in a single `if` -- but that might be a little hard to read).

Reply:
I used to use `continue`, but I started to move away from it to avoid issues regarding missing coverage. When `continue` is used, you have to be able to verify those negative cases as well. With the way the PR currently is, you only have to cover the one branch.